packages/blocks/src/api/categories.js

```diff
@@ -3,19 +3,32 @@
  */
 import { dispatch, select } from '@wordpress/data';
 
+/** @typedef {import('../store/reducer').WPBlockCategory} WPBlockCategory */
+
 /**
  * Returns all the block categories.
  *
- * @return {Object[]} Block categories.
+ * @return {WPBlockCategory[]} Block categories.
  */
 export function getCategories() {
 	return select( 'core/blocks' ).getCategories();
 }
 
+/**
+ * Returns a single category by slug.
+ *
+ * @param {string} slug Category slug.
+ *
+ * @return {WPBlockCategory|undefined} Block category, if exists.
+ */
+export function getCategory( slug ) {
```

Comment:
Can we avoid the new global functions, I'm thinking we should favor selectors and registry selectors as much as we can to avoid the "singleton" issue. We definitely have this issue in a lot of existing functions but having less global functions will force us to use selectors for these more.

Reply:
> Can we avoid the new global functions, I'm thinking we should favor selectors and registry selectors as much as we can to avoid the "singleton" issue. We definitely have this issue in a lot of existing functions but having less global functions will force us to use selectors for these more.

I agree with the general idea. For me, it was added purely for the sake of "consistency for consistency's sake", in that intentionally introducing inconsistency for when these global functions do and don't exist could likely lead to some confusion and uncertainty.

I don't feel particularly strongly about it, and can remove it if it's undesirable.

Reply:
I don't mind it being added for now but how can we discourage its usage in our code base.

Reply:
> I don't mind it being added for now but how can we discourage its usage in our code base.

Something like this, perhaps?

https://astexplorer.net/#/gist/1600f6a5862962ed4708045a7742dfeb/1be4a55a8de2a91b790352d1a1c01161c8d411a4

https://github.com/WordPress/gutenberg/blob/8c307ba/.eslintrc.js#L57

Reply:
In comparing the alternative approaches between #22192 and #22280, where the latter avoided the addition of a new selector, I wonder if it's something we can do here as part of the registration. Namely: Avoid `getCategory`, and let `registerBlockType` handle the category normalization.

Reply:
> In comparing the alternative approaches between #22192 and #22280, where the latter avoided the addition of a new selector, I wonder if it's something we can do here as part of the registration. Namely: Avoid `getCategory`, and let `registerBlockType` handle the category normalization.

Implemented in f1f59ee.